### DISCUSSION TITLE
dpgen for multi_systems

### DISCUSSION THREAD

**Discussion by yanjx001** [deepmodeling/dpgen#646] [Q&A]
Dear all,

I would like to run dpgen for the multi_systems, for example CxHy systems (x=0,1; y=2, 4), i.e. containing H2, CH2, CH4 three systems. Firstly, I generated and prepared the initial data in following three directories: H2.POSCAR.01x01x01/02.md/sys-0002/deepmd, CH2.POSCAR.01x01x01/02.md/sys-0002-0001/deepmd and CH4.POSCAR.01x01x01/02.md/sys-0004-0001/deepmd. In this step, I adopted the _"elements":["H"], ["H","C"], ["H","C"]_ with _"type_map" : ["H","C"]_ for each running of  "dpgen init_bulk PARAM". 
![image](https://user-images.githubusercontent.com/97779203/154832703-1f7fddcd-7238-47ff-b2dc-380056706793.png)

Then, I met some problems in param.json for main process. According to the descriptions, I added '"init_multi_systems": true,' and modified '  "init_data_prefix": "/path/to/init/"  "init_data_sys": [ "H2.POSCAR.01x01x01/02.md/sys-0002/deepmd", "CH2.POSCAR.01x01x01/02.md/sys-0002-0001/deepmd","CH4.POSCAR.01x01x01/02.md/sys-0004-0001/deepmd"], but the dpgen code cannot find the correct directories containing three systems. Anyone knows how to fix it? Any other more setting about "sys_configs" and training process?? Many thanks.

Best regards.
Wayne

**Comment by AnguseZhang** [comment]
`init_multi_systems` is used for "If set to true, init_data_sys directories should contain sub-directories of various systems. DP-GEN will regard all of these sub-directories as inital data systems"(See manual). 
You can first let `init_multi_systems` be False, try running `dpgen` command and post DP-GEN's error log here.

**Reply by yanjx001** [reply]
Thanks for your reply. DP-gen run smoothly now and I got 12 trained frozen_model.pb files (4 frozen_model.pb in iter.000000(1,2)/00.train/) in total. Which one should be chosen for next lammps calculations?

**Reply by AnguseZhang** [reply]
> 

You can use one of four models in the last iteration.  Statistically different four models with different random seeds should share a close training accuracy.
